src/modules/flight_mode_manager/FlightModeManager.cpp

```diff
@@ -533,17 +601,73 @@ void FlightModeManager::reset_setpoint_to_nan(vehicle_local_position_setpoint_s
 	setpoint.thrust[0] = setpoint.thrust[1] = setpoint.thrust[2] = NAN;
 }
 
-int FlightModeManager::task_spawn(int argc, char *argv[])
+FlightTaskError FlightModeManager::switchTask(FlightTaskIndex new_task_index)
 {
-	bool vtol = false;
+	// switch to the running task, nothing to do
+	if (new_task_index == _current_task.index) {
+		return FlightTaskError::NoError;
+	}
 
-	if (argc > 1) {
-		if (strcmp(argv[1], "vtol") == 0) {
-			vtol = true;
-		}
+	// Save current setpoints for the next FlightTask
+	vehicle_local_position_setpoint_s last_setpoint{};
```

Comment:
```suggestion
	vehicle_local_position_setpoint_s last_setpoint = FlightTask::empty_setpoint;
```

`empty_setpoint` contains NANs and not zeroes. It's always initialized with NANs such that it's clear which setpoints were explicitly set and which are not.

Reply:
Fixed in 59991ca